parser/quakec-lexer.js

```diff
@@ -54,9 +54,10 @@ const new_lexer = function() {
     lexer.addRule(/'\s*-?[0-9]+(\.[0-9]+)?\s+-?[0-9]+(\.[0-9]+)?\s+-?[0-9]+(\.[0-9]+)?\s*'/, process_lexeme("vector"));
     lexer.addRule(/#[0-9]+/, process_lexeme("builtin"));
     lexer.addRule(/\.?(void|float|vector|string|entity|\$frame)\b/, process_lexeme("type"));
-    lexer.addRule(/[A-Za-z_]+[A-Za-z0-9_]*/, process_lexeme("name"));
+    lexer.addRule(/[A-Za-z_$]+[A-Za-z0-9_]*/, process_lexeme("name"));
```

Comment:
I'm debating how to best handle this. `$xxx` is technically a valid name (the qcc compiler calls it a "frame macro"), and will work as a right-hand side (like `float x = $xxx;`), but they cannot be defined any other way (float $x gives "unknown frame macro $x"; attempting to use a valid frame macro as a name gives "not a name"). We need to error out on these.

Reply:
Done; directives can now not be used as names, same for frame macros